metrics/distances.py

```diff
@@ -0,0 +1,86 @@
+from votekit.profile import PreferenceProfile
+import numpy as np
+import ot
+import networkx as nx
+from typing import Union
+
+# TODO: Update em_dist to manually create graph in the function. Graph should be able to handle
+# or incomplete ballost
+
+
+def earth_mover_dist(
+    pp1: PreferenceProfile, pp2: PreferenceProfile, ballot_graph
+) -> int:
+    """
+    Computes the earth mover distance between two elections. \n
+    Assumes both elections share the same candidates
+    """
+    # Solving Earth Mover Distance
+    election_arrays = profilePairs_to_arrays(pp1, pp2)
+    electA_distr = election_arrays[0]
+    electB_distr = election_arrays[1]
+
+    # TODO:create ballot graph:
+
+    # Floyd Warshall Shortest Distance alorithm. Returns a dictionary of shortest path for each node
+    FW_dist_dict = nx.floyd_warshall(ballot_graph)
+    keysList = list(FW_dist_dict.keys())
+    keysList.sort()
+    cost_matrix = np.zeros((len(keysList), len(keysList)))
+    for i in range(len(keysList)):
+        node_dict = FW_dist_dict[keysList[i]]
+        cost_col = [value for key, value in sorted(node_dict.items())]
+        cost_matrix[i] = cost_col
+    G0 = ot.emd(electA_distr, electB_distr, cost_matrix)
+
+    # Hadamard Product = Earth mover dist between two matrices
+    earth_mover_dist = np.sum(np.multiply(cost_matrix, G0))
+    return earth_mover_dist
+
+
+def Lp_dist(
```

Comment:
tip: public functions should be lower case and have meaningful names

Reply:
done